### PR TITLE
Next.js migration

### PR DESCRIPTION
**Pull request by lfades**
Moved the HTML and CSS to Next.js pages, and the API routes for the Slack endpoints, the migration is ready and here are some key differences:

- All Slack API calls are using the new OAuth recommended by Slack except one, the only way to invite an user is by using an undocumented API and a legacy token, therefore the current setup still requires a legacy token, the only difference is that an OAuth token can be used for the other API calls
- The API section and iframe were removed
- The old implementation has been deleted

### CODE REVIEW

**Comment by ghost** [comment]
Slack subdomain is not defined

**Comment by lfades** [comment]
@tanahatas Can you expand more on what do you mean?.
To make the app work you need to follow the steps in the readme (the readme in this PR)

**Comment by ghost** [comment]
How to input my slack subdomain like blablabla.slack.com ?

Because your PR using https://${teamName}.slack.com and ${teamName} is undefined.

**Comment by lfades** [comment]
@tanahatas The team name is returned by `/api/v1/team`, which is fetched in `getInitialProps` to return the info of the team to the components that require `teamName`.

File: utils/slack.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import fetch from 'isomorphic-unfetch'
+
+/**
+ * Returns an absolute URL in the case of `getInitialProps`
+ */
+function getOrigin(req) {
+  if (typeof window === 'undefined') {
+    const { host } = req.headers
```
Comment (timneutkens) [inline utils/slack.js]: This is only safe when deployed to Now, on other hosting providers `host` can be overwritten by  the incoming request.

**Pull request closed without merging**